### PR TITLE
refactor: rename string module into text

### PR DESCRIPTION
**Pull request by PrettyWood**
Last PR for 0.11.0 on my side :)

- rename `string` module into `text` module
- rename `substring` function into `extract` to be even more explicit
- update doc

@max-sixty If you're ok with the PR feel free to merge

### CODE REVIEW

**Comment by vanillajonathan** [comment]
In all the languages I've came across its been `string`, `String` or `str` but never `text`.

In many languages I've come across `substr` or `substring` but never `extract`. When I see `substring` I know what it is, and what it does, but when I see `extract`, I don't know what that is, maybe that is extract, transform, load (ELT) ?

**Comment by PrettyWood** [comment]
@vanillajonathan for non technical people it should be way easier. This was the orignal will of @max-sixty and @snth. IMO it's just a question of habit and good documentation.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,8 +7,8 @@
 - _Breaking_: Exclude `group`'s `by` columns from the partition. See #3490
 - _Breaking_: `round` is now in the `math` module and needs to be called via
   `math.round`. See #3928
-- _Breaking_: `lower` and `upper` are now in the `string` module and need to be
-  called via `string.lower` and `string.upper`. See #3913
+- _Breaking_: `lower` and `upper` are now in the `text` module and need to be
+  called via `text.lower` and `text.upper`. See #3913
```
Comment (eitsupi) [inline CHANGELOG.md]:
```suggestion
  called via `text.lower` and `text.upper`. See #3913, #3973
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,9 +18,9 @@
   `cos`, `acos`, `sin`, `asin`, `tan`, `atan`, `pow` and `round`.\
   Those functions are in the `math` module (@PrettyWood, #3909, #3916 & 3928)
 - Most standard string functions are now supported: `ltrim`, `rtrim`, `trim`,
-  `length`, `substring`, `replace`. Utility functions `starts_with`, `contains`
+  `length`, `extract`, `replace`. Utility functions `starts_with`, `contains`
   and `ends_with` are also available.\
-  Those functions are in the `string` module (@PrettyWood, #3913)
+  Those functions are in the `text` module (@PrettyWood, #3913)
```
Comment (eitsupi) [inline CHANGELOG.md]:
```suggestion
  Those functions are in the `text` module (@PrettyWood, #3913, #3973)
```

**Comment by max-sixty** [comment]
> In all the languages I've came across its been `string`, `String` or `str` but never `text`.
> 
> In many languages I've come across `substr` or `substring` but never `extract`. When I see `substring` I know what it is, and what it does, but when I see `extract`, I don't know what that is, maybe that is extract, transform, load (ELT) ?

I empathize with the sentiment — but check out https://www.ablebits.com/office-addins-blog/excel-substring-functions-extract-text/ for Excel's approach.

FWIW I too was surprised at how folks who aren't traditional programmers don't know the term — but they don't...

I also think that `text` is a better term overall — we're not stringing together punch cards any longer...

**Pull request merged**